tests/sklearn/test_sklearn_models.py

```diff
@@ -1735,20 +1732,14 @@ def check_for_divergent_predictions(x, model, fhe, max_iterations=N_ALLOWED_FHE_
     fhe_diff_found = check_for_divergent_predictions(x, model, fhe="execute")
 
     # Check for differences in predictions
-    if is_linear_model:
-
-        # Linear models should give the same results whatever the p_error
-        assert fhe_diff_found
-
-        # Simulation and FHE differs with very high p_error on leveled circuit
-        # FIXME https://github.com/zama-ai/concrete-ml-internal/issues/4343
-        assert not simulation_diff_found
+    if error_param["p_error"] > 0.5:
+        assert (
+            fhe_diff_found and simulation_diff_found
+        ), "When p_error is high, predictions should differ in both FHE and simulation."
     else:
-        assert fhe_diff_found and simulation_diff_found, (
-            f"Predictions not different in at least one run.\n"
-            f"FHE predictions differ: {fhe_diff_found}\n"
-            f"SIMULATE predictions differ: {simulation_diff_found}"
-        )
+        assert not (
+            fhe_diff_found or simulation_diff_found
+        ), "When p_error is low, predictions should not differ in FHE or simulation."
```

Comment:
also, always good to print the actual values if the conditions are not meant, it's easier to debug

Reply:
Yes done.